engine/access/rest/common/error.go

```diff
@@ -56,3 +62,37 @@ func (e *Error) Status() int {
 func (e *Error) Error() string {
 	return e.err.Error()
 }
+
+// ErrorToStatusError converts an Access API error into a grpc status error. The input may either
+// be a status.Error already, or an access sentinel error.
+func ErrorToStatusError(err error) StatusError {
```

Comment:
each API has unique error reporting protocols (grpc vs rest vs websockets). Create API specific converters return the appropriate mapping from access sentinels